WMF Framework/ImageCacheController.swift

```diff
@@ -248,7 +248,7 @@ public final class ImageCacheController: CacheController {
         }
         let options = [kCGImageSourceShouldCache as String: NSNumber(value: true)] as CFDictionary
         guard let cgImage = CGImageSourceCreateImageAtIndex(source, 0, options) else {
-            return nil
+            return createImageForRejectedCGImage(with: data, mimeType: mimeType)
```

Comment:
I observed that when trying to create an image from the PDF file, `CGImageSourceCreateImageAtIndex` was failing (and returning `nil` here). My suspicion is that `CGImageSourceCreateImageAtIndex` probably worked fine in the past for PDFs (and possibly still does for some) but that Apple may well have changed something under the hood. So I added a recovery method to try and generate an image from the first page of the PDF (i.e. so it will ONLY fall through to this behaviour as a last resort).